### ISSUE TITLE
Any plans to create some sort of gisting framework for arbitrary models?

### ISSUE THREAD

**Issue by FFFiend** [jayelm/gisting#12]
Basically title, or excuse my ignorance if this is not possible. Right now it seems like only FLAN-T5 and LLama-7b are supported.

**Comment by jayelm** [comment]
In principle you should be able to apply gist masking to basically any
decoder or encoder-decoder Transformer language model (see the paper for
details), but you're right that this will require training your own model,
and in most cases, modifying the implementation of the LM to support custom
attention masks.

You can take a look at the implementations of `gist_llama` and `gist_t5` to
see how such masking is implemented, and diff my implementation with the
reference Transformers implementation, but right now unfortunately I don't
have plans to support other LMs. Sorry!

**Comment by FFFiend** [comment]
Thanks Jesse 😄
